File: parcours/src/main/java/fr/gaetanquenouille/parcours/DTO/SessionDTO.java
package fr.gaetanquenouille.parcours.DTO;

import java.time.LocalDateTime;
import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class SessionDTO {
    private Long id;
    private String label;
    private LocalDateTime beginsAt;
    private LocalDateTime endsAt;
    private List<String> fileUrls;
    private List<UserDTO> users;
    private List<SubjectDTO> subjects;
    private List<HomeworkDTO> homeworks;
}

File: frontend/src/app/model/session.dto.ts
import { UserDTO } from "./user.dto";
import { SubjectDTO } from "./subject.dto";

export interface SessionDTO {
    id: number;
    label: string;
    beginsAt: string; // ISO string format for LocalDateTime
    endsAt: string; // ISO string format for LocalDateTime
    users: UserDTO[];
    subjects: SubjectDTO[];
}